src/Agent.Worker/JobRunner.cs

```diff
@@ -279,6 +279,27 @@ public async Task<TaskResult> RunAsync(Pipelines.AgentJobRequestMessage message,
                     }
                 }
 
+#if OS_WINDOWS
+            bool retainDefaultEncoding = jobContext.Variables.GetBoolean(Constants.Variables.Agent.RetainDefaultEncoding) ?? false;
+            if (!retainDefaultEncoding) {
+                using (var p = HostContext.CreateService<IProcessInvoker>()) {
+                    int exitCode = p.ExecuteAsync(workingDirectory: HostContext.GetDirectory(WellKnownDirectory.Root),
+                                             fileName: WhichUtil.Which("chcp", true, Trace),
+                                             arguments: "65001",
+                                             environment: null,
+                                             cancellationToken: CancellationToken.None).GetAwaiter().GetResult();
```

Comment:
instead of doing `GetAwaiter().GetResult()`, you can just `int exitCode = await p.ExecuteAsync ()`